validation/immutable-references.go

```diff
@@ -0,0 +1,58 @@
+package validation
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// ContractASTsWithImmutableReferences stores the `immutableReferences`, copied from the compiled contract AST (from the combined JSON
+// artifact from the monorepo. We do this because the contracts and compiled artifacts are not available in the superchain
+// registry. Ex: ethereum-optimism/optimism/packages/contracts-bedrock/forge-artifacts/MIPS.sol/MIPS.json
+var ContractASTsWithImmutableReferences = map[string]string{
+	"AnchorStateRegistryProxy": `"immutableReferences":{"92829":[{"start":387,"length":32},{"start":828,"length":32},{"start":2296,"length":32}]}`,
+	"DelayedWETHProxy":         `"immutableReferences":{"97827":[{"start":831,"length":32},{"start":4133,"length":32}]}`,
+	"FaultDisputeGame":         `"immutableReferences":{"92829":[{"start":387,"length":32},{"start":828,"length":32},{"start":2296,"length":32}]}`,
+	"MIPS":                     `"immutableReferences":{"85798":[{"start":178,"length":32},{"start":1771,"length":32}]}`,
+}
```

Comment:
If possible, it would be great to store these in the `standard` package in a toml file, as we do for the versions and bytecodes themselves.

Furthermore, these references are _per op-contracts tag_, so we need to reflect that in the underlying data structures. So these need to be in a table under `op-contracts/v1.4.0`, with space for different references for future versions.

Reply:
Bonus points if it becomes clear to whoever reads the toml file, that the validation is masking the immutables (so is in some sense a slightly weaker validation than it is for other contracts which don't have immutables).